### DISCUSSION TITLE
Text file formatting standard

### DISCUSSION THREAD

**Discussion by 130s** [space-ros/space-ros#210] [Q&A]
Is `.rst` a requirement? If so where does it come from? `FRET`? docs.ros.org?
.rst might be more powerful, but much harder to write hence raises a bar much higher in my experience (I hope such difficulty wouldn't be the cause of stagnizing document e.g. https://github.com/space-ros/docs/issues/34).

**Comment by 130s** [comment]
Gemini didn't have a clear answer.

> Q. In space_ros project, is there a required text file format? .rst? I see more .rst in https://github.com/space-ros/ than .md files.
>
> A. You're correct! Based on the observation that the Space ROS documentation on GitHub uses mostly `.rst` files (reStructuredText), it's likely the preferred format for their documentation. 
> 
> Here's why you might see more `.rst` files:
> 
> * **Community Standard:** The ROS 2 documentation itself uses reStructuredText, so Space ROS, being derived from ROS 2, might have adopted the same format for consistency.
> * **Benefits of reStructuredText:** `.rst` offers advantages like better control over structure, easier integration with build systems, and potential for generating different output formats (e.g., HTML, PDF).
> 
> While `.md` (Markdown) files are also common for documentation, Space ROS might prefer the more structured approach of reStructuredText for their project.
> 
> Here are some resources to confirm and learn more:
> 
> * **Space ROS Documentation Site:**  [https://github.com/space-ros/space-ros](https://github.com/space-ros/space-ros) - This site doesn't explicitly state the format preference, but the file extensions are a good indicator.
> * **ROS 2 Documentation:** [https://docs.ros.org/en/foxy/index.html](https://docs.ros.org/en/foxy/index.html) - This confirms ROS 2 uses reStructuredText.
> 
> If you need to contribute documentation to Space ROS, it's best to check their contribution guidelines or reach out to the maintainers for confirmation.

**Reply by franklinselva** [reply]
reStructured Text has some benefits which markdown lacks. The way I have been using `.rst` is to build websites directly from the docs of the repositories. Even though the docs section sometimes can look like templates than actual content.

Few areas I could think of at the moment, 
 - If there are examples that need to be built and confirmed it's working,`.rst` is the go-to option since we can reference the programs directly / reference a section with line numbers of the program.
 - The `.rst` can be compiled into HTML for better anchoring, table of contents and other output formats as mentioned
 - There is good interoperability between `.rst` and `.md` since `.rst` is one of the preferred formats for documentation tools like sphinx
 - I think hierarchy of documentation is easier

**Reply by 130s** [reply]
Sure, feature-wise, .rst might be more powerful as I wrote in OP. I am also a fan of it.

But, I've seen over years the downsides of it discouraged its usage in some communities esp. time-sensitive environment e.g. startup company. For those cases I've seen, higher learning curve and difficulty in writing were bigger issues than the feature-richness.

I'm just asking, not suggesting to switch tools. But if there's a room for switching I'll be glad that I opened this question :)

**Reply by franklinselva** [reply]
Valid point. Let's hear what others have to say.

**Comment by Bckempa** [comment]
Can you clarify what you mean by "requirement" in your question? It isn't in the [contributing guide](https://github.com/space-ros/.github/blob/master/CONTRIBUTING.md) (itself a markdown document) or any of the primary code repositories.

reStructured Text is used for website generation ([space-ros/spaceros.org](https://github.com/space-ros/spaceros.org) and [space-ros/docs](https://github.com/space-ros/docs)) for basically all the reasons you got from the LLM - it is [arguably better](https://buttondown.com/hillelwayne/archive/why-i-prefer-rst-to-markdown/) when you want more layout control than basic Markdown documents, the Space ROS documentation site is based off the ROS 2 documentation site which also uses .rst sources, and it is the default markup language of the Python ecosystem the website generation tools ([Pelican](https://getpelican.com) and [Sphinx](https://www.sphinx-doc.org/en/master/), respectively) come from.

We take the issue of outdated documentation seriously, but all the stale bits I've seen were a direct results of the standard issues in early-stage open source projects, namely limited bandwidth by a team of volunteers. Most of the documentation that needs to be updated was written by paid contributors during the ACO who are no longer on the project.

I am not aware of anyone who attempted to update the documentation and was unable to due to the reStructured syntax and migrating to a different stack (which wouldn't benefit from having the ROS docs as an upstream source of updates) would increase the maintenance burden of the entire documentation system so I am skeptical of that as a solution.
